lib/sinon/sandbox.js

```diff
@@ -95,6 +95,10 @@ extend(sinonSandbox, {
     },
 
     restore: function () {
+        if (arguments.length) {
+            throw new Error("sandbox.restore() does not take any parameters. Perhaps you meant stub.restore()");
+        }
+
         sinonCollection.restore.apply(this, arguments);
```

Comment:
This line is weird to me. collection.restore also doesn't take any arguments, so it seems weird to be passing through this functions arguments when neither of them take any. 

I considered removing it here but decided that I didn't have enough context and didn't want to make this PR any more risky to you guys by changing existing code.